### PR TITLE
Added strict types [Review module]

### PR DESCRIPTION
**Pull request by ghost**
### Description
- Small PHPDocs fixes in Review module
- Added strict types

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by ihor-sviziev** [comment]
Hi @SikailoISM,
Your changes looks good, but here is one unit test is failing. Could you fix it?

**Comment by ghost** [comment]
> Hi @SikailoISM,
> Your changes looks good, but here is one unit test is failing. Could you fix it?

OK, I will fix it.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4343](https://jira.corp.magento.com/browse/ENGCOM-4343) has been created to process this Pull Request

File: app/code/Magento/Review/Model/Rating/Entity.php [hunk 1]
```diff
@@ -28,7 +31,7 @@ protected function _construct()
      * @param string $entityCode
      * @return int
      */
-    public function getIdByCode($entityCode)
+    public function getIdByCode(string $entityCode): int
```
Comment (sivaschenko) [inline app/code/Magento/Review/Model/Rating/Entity.php:34]: Please see the actual implementation - the return value is retrieved using \Magento\Framework\DB\Adapter\AdapterInterface::fetchOne that actually returns a string - so the declared return strict type results in a fatal error.

Please ensure strict types are matching the actual return values

File: app/code/Magento/Review/Model/Rating.php [hunk 2]
```diff
@@ -145,9 +149,9 @@ public function getEntitySummary($entityPkValue, $onlyForCurrentStore = true)
     /**
      * @param int $reviewId
      * @param bool $onlyForCurrentStore
-     * @return array
+     * @return Rating
      */
-    public function getReviewSummary($reviewId, $onlyForCurrentStore = true)
+    public function getReviewSummary(int $reviewId, bool $onlyForCurrentStore = true): Rating
```
Comment (sivaschenko) [inline app/code/Magento/Review/Model/Rating.php:154]: Please remove param strict types to ensure backward compatibility

File: app/code/Magento/Review/Model/Rating.php [hunk 3]
```diff
@@ -134,9 +138,9 @@ public function getOptions()
      *
      * @param int $entityPkValue
      * @param bool $onlyForCurrentStore
-     * @return \Magento\Framework\Data\Collection\AbstractDb
+     * @return array
      */
-    public function getEntitySummary($entityPkValue, $onlyForCurrentStore = true)
+    public function getEntitySummary(int $entityPkValue, bool $onlyForCurrentStore = true): array
```
Comment (sivaschenko) [inline app/code/Magento/Review/Model/Rating.php:143]: Please remove param strict types to ensure backward compatibility

File: app/code/Magento/Review/Model/Rating.php [hunk 4]
```diff
@@ -97,7 +101,7 @@ public function addOptionVote($optionId, $entityPkValue)
      * @param int $optionId
      * @return $this
      */
-    public function updateOptionVote($optionId)
+    public function updateOptionVote(int $optionId): Rating
```
Comment (sivaschenko) [inline app/code/Magento/Review/Model/Rating.php:104]: Please remove param strict types to ensure backward compatibility (i.e. $optionId can be currently passed as a string by third-party extension)

File: app/code/Magento/Review/Model/Rating.php [hunk 5]
```diff
@@ -79,7 +83,7 @@ protected function _construct()
      * @param int $entityPkValue
      * @return $this
      */
-    public function addOptionVote($optionId, $entityPkValue)
+    public function addOptionVote(int $optionId, int $entityPkValue): Rating
```
Comment (sivaschenko) [inline app/code/Magento/Review/Model/Rating.php:86]: Please remove param strict types to ensure backward compatibility

File: app/code/Magento/Review/Helper/Data.php [hunk 6]
```diff
@@ -51,7 +53,7 @@ public function __construct(
      * @param string $origDetail
      * @return string
      */
-    public function getDetail($origDetail)
+    public function getDetail(string $origDetail): string
```
Comment (sivaschenko) [inline app/code/Magento/Review/Helper/Data.php:56]: Please remove param strict types to ensure backward compatibility (i.e. $origDetail can be currently passed as a \Magento\Framework\Phrase by third-party extension)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @SikailoISM thanks for the pull request, please see my review notes

**Comment by sidolov** [comment]
@SikailoISM , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**